### PR TITLE
PR: Create a new Appearance plugin

### LINKED ISSUE #12798: Create new Appearance Plugin using new API
- [ ] Move the appearance preferences to a separate plugin.
- [ ] Provide API for theme and icon extension

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #12798

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by goanpeca** [comment]
Hi @dalthviz, this one is ready for QA testing. Thanks!

**Comment by dalthviz** [comment]
Not sure why but seems like tests are failing on Windows @goanpeca (a message about Python aborting appears)

**Comment by goanpeca** [comment]
Yeah @dalthviz I am not sure where it is coming from, but please test the PR as the error seems unrelated to the work.

**Comment by dalthviz** [comment]
Checking this I found that changing the interface theme option (if no need of restart is detected) leads to the following traceback:

```python
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\preferences\configdialog.py", line 227, in accept
    configpage.apply_changes()
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\preferences\configdialog.py", line 99, in apply_changes
    self.apply_callback()
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\preferences\configdialog.py", line 282, in <lambda>
    self.apply_settings(self.changed_options))
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\plugins\appearance\confpage.py", line 234, in apply_settings
    self.main.help.apply_plugin_settings(['color_scheme_name'])
AttributeError: 'Help' object has no attribute 'apply_plugin_settings'

```

Besides that seems like the rest of the options are working :+1: . Also, the error above is present in master too.

As a side note, restarting Spyder while running it from bootstrap is raising an error (this is also happening in master and 4.x):

```python
Traceback (most recent call last):
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\app\restart.py", line 32, in <module>
    from spyder.utils.qthelpers import qapplication
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\utils\qthelpers.py", line 28, in <module>
    from spyder.config.manager import CONF
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\config\manager.py", line 17, in <module>
    from spyder.config.main import CONF_VERSION, DEFAULTS, NAME_MAP
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\config\main.py", line 19, in <module>
    from spyder.config.fonts import MEDIUM, SANS_SERIF
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\config\fonts.py", line 14, in <module>
    from spyder.config.utils import is_ubuntu
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\gonzalo\spyder\spyder\config\utils.py", line 17, in <module>
    from spyder_kernels.utils import iofuncs
ModuleNotFoundError: No module named 'spyder_kernels'

```

**Comment by goanpeca** [comment]
Thanks @dalthviz. Will fix!

**Comment by goanpeca** [comment]
>As a side note, restarting Spyder while running it from bootstrap is raising an error (this is also happening in master and 4.x):

Please open a new issue for that with any details you can add. Thanks, @dalthviz !

**Comment by goanpeca** [comment]
>Checking this I found that changing the interface theme option (if no need of restart is detected) leads to the following traceback:

I fixed (I think) the first issue @dalthviz mentioned. If you could check again that would be great.

Thanks!

**Comment by goanpeca** [comment]
This is ready for review @ccordoba12

**Comment by dalthviz** [comment]
@goanpeca checking again seems like now the help plugin doesn't raises an error :+1:

However, I checked the options regarding the fonts options and found the following:

* Seems like the `Plain text` option, even after changing it, puts as selected font `Consolas` in the combobox.
* Seems like the `Size` option of `Rich text` is connected to both number selectors (`Plain text` and `Rich text`)

font

**Comment by dalthviz** [comment]
Also, seems like the order of the sections in the preferences changed (now the Appearance entry is the seven one in the list instead of the second one)? (Maybe is not too important but I mention it just in case since I notice it while checking the GIF and running 4.x).

**Comment by goanpeca** [comment]
>Seems like the Plain text option, even after changing it, puts as selected font Consolas in the combobox.
Seems like the Size option of Rich text is connected to both number selectors (Plain text and Rich text)

Thanks @dalthviz, will fix.

>Also, seems like the order of the sections in the preferences changed (now the Appearance entry is the seven one in the list instead of the second one)? (Maybe is not too important but I mention it just in case since I notice it while checking the GIF and running 4.x).

Yes this will change in the meantime but will be restored once the other plugins are migrated. Thanks for checking!

**Comment by goanpeca** [comment]
>Seems like the Plain text option, even after changing it, puts as selected font Consolas in the combobox.
Seems like the Size option of Rich text is connected to both number selectors (Plain text and Rich text)

Fixed @dalthviz

**Comment by goanpeca** [comment]
@dalthviz @ccordoba12 this one is ready for review.

**Comment by ccordoba12** [comment]
@goanpeca, please rebase this one to see if our tests pass (they are failing in a Pylint test, which seems unrelated to this).

@dalthviz, please give this one another round of manual testing.

**Comment by goanpeca** [comment]
Fixed.

**Comment by ccordoba12** [comment]
@goanpeca , now this one has merge conflicts.

**Comment by goanpeca** [comment]
Fixed.

**Comment by ccordoba12** [comment]
I'll left this one to @dalthviz to test it and merge it.

**Comment by dalthviz** [comment]
While testing this I found the following:

* Seems like the elements in the Preference dialog are duplicated (although maybe this is happening beside this PR?):

image

The actual functionality of the Appearance section LGTM :+1:

**Comment by goanpeca** [comment]
>While testing this I found the following:

Hi @dalthviz thanks for checking.

I had already opened https://github.com/spyder-ide/spyder/pull/13414 to fix that. Feel free to merge that one too if you agree with the changes.

**Comment by dalthviz** [comment]
Done @goanpeca , please rebase this one and then it should be ready to merge :+1:

**Comment by goanpeca** [comment]
>Done @goanpeca , please rebase this one and then this should be ready for merge 👍

Fixed! Thanks again.

File: spyder/app/mainwindow.py [hunk 1]
```diff
@@ -1182,6 +1181,12 @@ def create_edit_action(text, tr_text, icon):
         self.run = Run(self, configuration=CONF)
         self.register_plugin(self.run)
 
+        # Appearance plugin
+        CONF.set('appearance', 'css_path', css_path)
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py:1185]: Why is this an option of `appearance` and not of `help`? I say it because `css_path` is a path inside Help.

Comment (goanpeca) [reply spyder/app/mainwindow.py]: Fixed

File: spyder/config/main.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 from spyder.plugins.editor.utils.findtasks import TASKS_PATTERN
 from spyder.utils.introspection.module_completion import PREFERRED_MODULES
 
-
+#
```
Comment (dalthviz) [inline spyder/config/main.py:25]: pep8speaks issue:

* In the file [`spyder/config/main.py`](https://github.com/spyder-ide/spyder/blob/109e236470b80ca588d5f7c0894ec455dad65014/spyder/config/main.py):

> [Line 25:2](https://github.com/spyder-ide/spyder/blob/109e236470b80ca588d5f7c0894ec455dad65014/spyder/config/main.py#L25): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

Comment (goanpeca) [reply spyder/config/main.py]: Fixed.

**Pull request merged**